crates/sargon/src/system/sargon_os/sargon_os.rs

```diff
@@ -151,6 +151,38 @@ impl SargonOS {
         Ok(())
     }
 
+    pub async fn derive_wallet(
```

Comment:
Why "derive"? This is not deriving anything..?

Why "wallet"? 

Can we name it something "profile with accounts" ?

Reply:
Wanted to keep the wallet name consistent with the other "entry" methods like

* `new_wallet`
* `import_wallet`
* `delete_wallet`

Reply:
Unfortunately that is impossible. I wanted to do the same, but in order to do that I need another biometric prompt. The same should apply to import wallet. 

Why is that?
* If we need to ask for biometrics only once, we need to pass the mnemonic to ffi. I am not sure if we want to do that. If we ever did that, then sargon os could check the validity and store it. Thus having total control over it.
* Otherwise we let the host store the mnemonic, as it used to, but now this method will need to read the mnemonic again, asking for biometrics for a second time.

Reply:
Right, I think we should do any of those too.

We are already passing the mnemonic across FFI in the form of entropy. So I think best pass a PrivateHierarchicalDeterministicFactorSource from Host, that way we **know** user has intact "wallet"

Reply:
Or well, we know if we also check that the accounts are controlled by **that** DeviceFactorSource

Reply:
I am in favour of passing the mnemonic over FFI, unless there are some specific concerns. Can't see any in particular.

Reply:
> We are already passing the mnemonic across FFI in the form of entropy.

You are right @CyonAlexRDX I forgot about that. Will make the changes.

Reply:
@CyonAlexRDX I have implemented the logic to 
1. pass the`PrivateHierarchicalDeterministicFactorSource` into the function 
2. Validate the accounts public keys
3. Ensure that secure storage succeeds saving the mnemonic
 https://github.com/radixdlt/sargon/pull/212/commits/84c7af6878f4e81aeb55eda43049b6df5bf8f6e3
 
 Please let me know if the execution is valid